docs/source/about_mapstyle_vs_iterable.mdx

```diff
@@ -0,0 +1,220 @@
+# Differences between Dataset and IterableDataset
+
```

Comment:
Maybe just add a sentence or two here that introduces the topic and scope of the doc. Something like:

There are two types of dataset objects, a `Dataset` and an `IterableDataset`. Whichever type of dataset you choose to use or create depends on the size of the dataset. In general, an `IterableDataset` is ideal for big datasets (think hundreds of GBs!) due to its lazy behavior and speed advantages, while a `Dataset` is great for everything else. This page will compare the differences between a `Dataset` and an `IterableDataset` to help you pick the right dataset object for you.